pkgs/development/compilers/rust/cargo.nix

```diff
@@ -21,7 +21,9 @@ rustPlatform.buildRustPackage {
   buildInputs = [ cacert file curl python3 openssl zlib libgit2 ]
     ++ stdenv.lib.optionals stdenv.isDarwin [ CoreFoundation Security libiconv ];
 
-  LIBGIT2_SYS_USE_PKG_CONFIG = 1;
+  # cargo uses git-rs which is made for a version of libgit2 from recent master that
+  # is not compatible with the current version in nixpkgs.
+  #LIBGIT2_SYS_USE_PKG_CONFIG = 1;
```

Comment:
You can remove `libgit2` from buildInputs in that case.

Reply:
Good point, fixed.